hybrasyl/World.cs

```diff
@@ -1963,6 +2018,111 @@ private void PacketHandler_0x2D_PlayerInfo(Object obj, ClientPacket packet)
+            // additional work that needs to happen though I haven't been able to determine
+            // what those bytes represent yet...
+
+            if (partner == null)
+            {
+                return;
+            }
+
+            switch (stage)
+            {
+                // Stage 0x02 means that a user is sending an initial request to the invitee.
+                // That means we need to check whether the user is a valid candidate for
+                // grouping, and send the confirmation dialog if so.
+                case 0x02:
+                    Logger.InfoFormat("{0} invites {1} to join a group.", user.Name, partner.Name);
+                    
+                    // Remove the user from the group. Kinda logically weird beside all of this other stuff 
+                    // so it may be worth restructuring but it should be accurate as-is.
+                    if (partner.Grouped && user.Grouped && partner.Group == user.Group)
+                    {
+                        Logger.InfoFormat("{0} leaving group.", user.Name);
+                        user.Group.Remove(partner);
+                        return;
+                    }
+
+                    // Now we know that we're trying to add this person to the group, not remove them.
+                    // Let's find out if they're eligible and invite them if so.
+                    if (partner.Grouped)
+                    {
+                        user.SendMessage(String.Format("{0} is already in a group.", partner.Name), MessageTypes.SYSTEM);
+                        return;
+                    }
+
+                    if (!partner.Grouping)
+                    {
+                        user.SendMessage(String.Format("{0} is not accepting group invitations.", partner.Name), MessageTypes.SYSTEM);
+                        return;
+                    }
+
+                    // Send partner a dialog asking whether they want to group (opcode 0x63).
+                    ServerPacket response = new ServerPacket(0x63);
+                    response.WriteByte((byte)0x01);
+                    response.WriteString8(user.Name);
+                    response.WriteByte(0);
+                    response.WriteByte(0);
+
+                    partner.Enqueue(response);
+                    break;
+                // Stage 0x03 means that the invitee has responded with a "yes" to the grouping
+                // request. We need to add them to the original user's group. Note that in this
+                // case the partner sent the original invitation.
+                case 0x03:
+                    Logger.Info("Invitation accepted. Grouping.");
+                    partner.InviteToGroup(user);
+                    break;
+                // This shouldn't happen but we should log it and fix it if it does.
+                default:
+                    Logger.Info("Unknown GroupRequest stage. No action taken.");
+                    break;

```

Comment:
All of these Info level logs should be going to debug instead, I think, in order to not clutter up the logfile. The Unknown stage should also be at Error level.